src/app/core/parsers/zod-entity-parser.service.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { ZodEntityParser } from './zod-entity-parser.service'
+import { StudioConfiguration } from '../../shared/services/studio-configuration'
+import { Tv2Action, Tv2ActionContentType } from '../../shared/models/tv2-action'
+import { Media } from '../../shared/services/media'
+
+describe(ZodEntityParser.name, () => {
+  describe(ZodEntityParser.prototype.parseStudioConfiguration.name, () => {
+    it('parses a studio configuration with url on http protocol and hostname', () => {
+      const testee: ZodEntityParser = new ZodEntityParser()
+      const expectedStudioConfiguration: StudioConfiguration = {
+        settings: {
+          mediaPreviewUrl: 'http://some.url',
+        },
+        blueprintConfiguration: {
+          ServerPostrollDuration: 123,
+        },
+      }
+      const parseStudioConfiguration: StudioConfiguration = testee.parseStudioConfiguration(expectedStudioConfiguration)
+      expect(parseStudioConfiguration).toEqual(expectedStudioConfiguration)
+    })
+
+    it('does not parse a studio configuration with empty url', () => {
+      const testee: ZodEntityParser = new ZodEntityParser()
+      const expectedStudioConfiguration: StudioConfiguration = {
+        settings: {
+          mediaPreviewUrl: '',
+        },
+        blueprintConfiguration: {
+          ServerPostrollDuration: 123,
+        },
+      }
+      const result = (): StudioConfiguration => testee.parseStudioConfiguration(expectedStudioConfiguration)
+      expect(result).toThrow()
+    })
+
+    it('does not parse a studio configuration with no host, just protocol', () => {
+      const testee: ZodEntityParser = new ZodEntityParser()
+      const expectedStudioConfiguration: StudioConfiguration = {
+        settings: {
+          mediaPreviewUrl: 'http://',
+        },
+        blueprintConfiguration: {
+          ServerPostrollDuration: 123,
+        },
+      }
+      const result = (): StudioConfiguration => testee.parseStudioConfiguration(expectedStudioConfiguration)
+      expect(result).toThrow()
+    })
+
+    it('does not parse a studio configuration with protocol and just one letter host', () => {
+      const testee: ZodEntityParser = new ZodEntityParser()
+      const expectedStudioConfiguration: StudioConfiguration = {
+        settings: {
+          mediaPreviewUrl: 'http://a',
+        },
+        blueprintConfiguration: {
+          ServerPostrollDuration: 123,
+        },
+      }
+      const result = (): StudioConfiguration => testee.parseStudioConfiguration(expectedStudioConfiguration)
+      expect(result).toThrow()
+    })
+  })
+
+  describe(ZodEntityParser.prototype.parseTv2Action.name, () => {
+    it('parses a Tv2Adtion ', () => {
```

Comment:
Tv2Action is misspelled.

And same note as the previous comment, that this is organisation specific logic.